hooks/_common.sh

```diff
@@ -445,6 +445,46 @@ function common::colorify {
   echo -e "${COLOR}${TEXT}${RESET}" >&2
 }
 
+#######################################################################
+# Set terraform binary path
+# Allows user to set the path to custom Terraform or OpenTofu binaries
+# Outputs:
+#   If failed - exit 1 with error message about missing Terraform/Opentofu binary
+function common::get_tf_binary {
+  local hook_config_tf_binary
+
+  for config in "${HOOK_CONFIG[@]}"; do
+    if [[ $config == tf_binary=* ]]; then
+      hook_config_tf_binary="${config#*=}"
+      break
+    fi
+  done
+
+  # direct hook config, has the highest priority
+  if [[ -n "${hook_config_tf_binary}" ]]; then
+    echo "${hook_config_tf_binary}"
```

Comment:
- Curly brackets are redundant where there's no value expansion or substitution is performed 
- Double quotes inside double square brackets are redundant generally
- `-n` is implied within double square brackets
- From my point of view it makes sense to `return` immediately when the condition is true
```suggestion
  if [[ $hook_config_tf_binary ]]; then
    echo "$hook_config_tf_binary"
    return
```

This applies to all conditionals within this function. Please adjust.

Reply:
I'm not entirely sure, though would it make sense to check whether target binary is executable and fail otherwise to fail early and to provide descriptive error to a user instead of relaying this to where the TF binary is executed actually? 🤔

Reply:
Also would it make sense to export `TF_PATH` var (it's what `TF_BINARY` is in your version) to replace var assignments in each hook that depends on it? What I mean is we could add call to this function into e.g. `common::initialize` to make this sort of global thing instead of adding this into each place where it is required. What do you think? 🤔